bpo-46066: Deprecate kwargs syntax for TypedDict definitions (#31126)
python/cpython: Doc/library/typing.rst

   ``Point2D.__total__``, ``Point2D.__required_keys__``, and
   ``Point2D.__optional_keys__``.
   To allow using this feature with older versions of Python that do not

>    support :pep:`526`, ``TypedDict`` can be created using a functional form:

Reviewer: I'm not keen on the repetition of the word "using" here, and with the bullet points I think we can go back to something similar to the existing text:

```suggestion
   support :pep:`526`, ``TypedDict`` supports two additional
   syntactic forms::
```

   ``Point2D.__optional_keys__``.
   To allow using this feature with older versions of Python that do not
   support :pep:`526`, ``TypedDict`` supports two additional equivalent
   syntactic forms.

>    Using dict as the second argument::

Reviewer: Why the line break here? You need two line breaks if you want to start a new paragraph.

```suggestion
   syntactic forms. Firstly, using a literal :class:`dict` as the
   second argument::
```

      Point2D = TypedDict('Point2D', {'x': int, 'y': int, 'label': str})
   

>    Using keyword arguments::

Reviewer: ```suggestion
   Secondly, using keyword arguments::
```

   
      Point2D = TypedDict('Point2D', x=int, y=int, label=str)

   .. deprecated-removed:: 3.11 3.13

>       The keyword arguments syntax is deprecated in 3.11 and will be removed
>       in 3.13. Currently it might not be supported by typecheckers.

Reviewer: ```suggestion
      The keyword-argument syntax is deprecated in 3.11 and will be removed
      in 3.13. It may also be unsupported by third-party type-checking tools.
```

   ``Point2D.__optional_keys__``.
   To allow using this feature with older versions of Python that do not
   support :pep:`526`, ``TypedDict`` supports two additional equivalent
   syntactic forms. Firstly, using a literal :class:`dict` as the

>    second argument::

Reviewer 2: I'd suggest just writing "a ``TypedDict`` may be created using a functional form". This parallels https://docs.python.org/3.10/library/enum.html#functional-api

Author: Thanks for this!

      Point2D = TypedDict('Point2D', {'x': int, 'y': int, 'label': str})

>    Secondly, using keyword arguments::

Reviewer 2: ```suggestion
   Keyword arguments may also be used::
```

      Point2D = TypedDict('Point2D', x=int, y=int, label=str)

   .. deprecated-removed:: 3.11 3.13
      The keyword-argument syntax is deprecated in 3.11 and will be removed

>       in 3.13. It may also be unsupported by third-party type-checking tools.

Reviewer 2: ```suggestion
      in 3.13. It may also be unsupported by static type checkers.
```
This is the standard term used in the documentation

Author: Done!

   support :pep:`526`, ``TypedDict`` supports two additional equivalent
   syntactic forms::

      Point2D = TypedDict('Point2D', {'x': int, 'y': int, 'label': str})
      Point2D = TypedDict('Point2D', x=int, y=int, label=str)

>    .. deprecated-removed:: 3.11 3.13

Reviewer 2: To make it more clear that you're only talking about one of the two alternative syntaxes, I think it makes sense to put both of them in separate example blocks.

Also, I'm not sure it makes sense to specifically mention mypy here. There is only one other place in this file that specifically mentions mypy.

Author: Done!
